src/packagepath.ts

```diff
@@ -13,7 +13,9 @@ let juliaPackagePath: string = null;
 export async function getPkgPath() {
     if (juliaPackagePath == null) {
         let jlexepath = await juliaexepath.getJuliaExePath();
-        var res = await exec(`"${jlexepath}" --startup-file=no --history-file=no -e "println(Pkg.dir())"`);
+        // TODO: there's got to be a better way to do this.
+        var res = await exec(`"${jlexepath}" --startup-file=no --history-file=no -e "(using Pkg;println(dirname([u[1][String([0x70,0x61,0x74,0x68])] for (p,u) in Pkg.Types.Context().env.manifest if haskey(u[1], String([0x70,0x61,0x74,0x68]))][1])))"`);
+        
```

Comment:
I feel we should just return ``.julia/dev`` here? @KristofferC, I assume that path is not hard coded but that we should query that somehow? Could you give us a pointer?

Reply:
What is this supposed to do?

Reply:
This is used for the "open package in new window" command. I think we just want to show a list of all the sub folders of ``./julia/dev`` as an option to open in a new window. So really, all we need to know is: should we hardcode the path ``.julia/dev`` here, or can that be configured by users, and if so, how can we figure out what a user configured?

Reply:
The correct way to me (which unfortunately uses some internal stuff) would be:

```jl
import Pkg
import UUIDs

function get_dev_list()::vector{pair{string, string}}
    ctx = pkg.types.context()
    devs = pair{string, string}[]
    # extract packages which are devved (contains a path entry in the manifest)
    for (pkgname, uuid) in ctx.env.project["deps"]
        info = pkg.types.manifest_info(ctx.env, uuids.uuid(uuid))
        if info !== nothing
            if haskey(info, "path")
                push!(devs, pkgname => abspath(dirname(ctx.env.project_file), info["path"]))
            end
        end
    end
    return devs
end
```

used as

```jl
julia> get_dev_list()
6-element Array{Pair{String,String},1}:
      "IterTools" => "/Users/kristoffer/.julia/dev/IterTools"
         "PkgDev" => "/Users/kristoffer/.julia/dev/PkgDev"
          "Yeppp" => "/Users/kristoffer/.julia/dev/Yeppp"
   "PkgBenchmark" => "/Users/kristoffer/.julia/dev/PkgBenchmark"
 "BaseBenchmarks" => "/Users/kristoffer/.julia/dev/BaseBenchmarks"
        "Pardiso" => "/Users/kristoffer/.julia/dev/Pardiso"
```

This also works if we dev a package at an arbitrary path

```jl
shell> pwd
/Users/kristoffer

shell> git clone https://github.com/JuliaLang/Example.jl
Cloning into 'Example.jl'...
...

(v1.1) pkg> dev Example.jl
 Resolving package versions...
  Updating `~/.julia/environments/v1.1/Project.toml`
  [7876af07] + Example v0.5.1+ [`~/.julia/environments/v1.1/../../../Example.jl`]
  Updating `~/.julia/environments/v1.1/Manifest.toml`
  [7876af07] + Example v0.5.1+ [`~/.julia/environments/v1.1/../../../Example.jl`]

julia> get_dev_list()
7-element Array{Pair{String,String},1}:
...
        "Example" => "/Users/kristoffer/Example.jl"
...
```

Reply:
The existing code does as suggested by @KristofferC (i.e. uses the loaded manifest). It looks revolting because I don't know how to use quotes _inside_ quotes, hence construction of strings from `UInt8`s...  :sob: